classes/install/Installer.php

```diff
@@ -1008,6 +1010,34 @@ public function checkPhpVersion()
         $this->setError(self::INSTALLER_ERROR_GENERAL, 'installer.unsupportedPhpError');
         return false;
     }
+
+    /**
+     * Add the app key if not already set
+     *
+     * @return bool Success/failure
+     */
+    public function addAppKey()
+    {
+        // if APP KEY already exists, nothing to do
+        if (PKPAppKey::hasKey()) {
+            return true;
+        }
+
+        // will set an error if app key variable not set
+        // but will not halt the process
+        if (!PKPAppKey::hasKeyVariable()) {
+            error_log("No key variable named `app_key` defined in the `general` section of config file. Please update the config file's general section and add line `app_key = `");
+            return true;
+        }
+
+        try {
+            PKPAppKey::writeToConfig(PKPAppKey::generate());
+        } catch (Throwable $exception) {
+            error_log($exception->getMessage());
+        } finally {
+            return true;
+        }
+    }
```

Comment:
If `app_key` writing to config file we are not halting the process as not making the app key a must but if we decide to make it a must, we can set error and return false to halt the process .